app/src/main/java/com/worldsnas/starplayer/model/RepositoryImpl.kt

```diff
@@ -0,0 +1,30 @@
+package com.worldsnas.starplayer.model
+
+import android.util.Log
+import com.worldsnas.starplayer.api.MusicResponse
+import com.worldsnas.starplayer.api.WebServiceApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class RepositoryImpl @Inject constructor(
+    private val localMusicProvider: LocalMusicProvider,
+    private val webServiceApi: WebServiceApi
+) : Repository {
+
+    override suspend fun getApiData(): List<MusicResponse> {
+
+        return withContext(Dispatchers.IO) {
```

Comment:
fek konam khode retrofit mibare ro `IO` va niazi nabashe ma bezanim
search konid age intor hast ino hazf konim

Reply:
doroste inja behesh eshare shode 
https://stackoverflow.com/questions/58083309/coroutines-and-retrofit-best-way-to-handle-errors